packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js

```diff
@@ -491,21 +495,64 @@ angular.module('managerApp').controller(
         .$promise.then(() => {
           this.hasIncidentOccured = true;
         })
-        .catch((error) => {
-          const errorStatus = get(error, 'status');
+        .catch((err) => {
+          const errorStatus = get(err, 'status');
           if (errorStatus === XDSL_NO_INCIDENT_CODE) {
             this.hasIncidentOccured = false;
           } else {
             this.TucToast.error(
               `${this.$translate.instant(
                 'xdsl_details_diagnostic_get_incident_error',
-              )} ${get(error, 'data.message', '')}`,
+              )} ${get(err, 'data.message', '')}`,
             );
           }
         })
         .finally(() => {
           this.$scope.loaders.incident = false;
         });
     }
+
+    /**
+     * Retrieve modem detail for exchange if available
+     */
+    getModemExchange() {
+      return this.OvhApiXdsl.Modem()
+        .v6()
+        .getComfortExchange({
+          xdslId: this.$stateParams.serviceName,
+        })
+        .$promise.then((result) => {
+          this.$scope.access.xdsl.canExchange = true;
+          this.$scope.access.xdsl.newModel = result.newModel
+            .replace(/\./g, ' ')
+            .toUpperCase();
+          this.$scope.access.xdsl.priceHT = result.price.value;
+          this.$scope.access.xdsl.priceTTC = result.priceWithTax.value;
+        })
+        .catch((error) => {
+          if (error.data.message.includes(XDSL_EXCHANGE_MODEM.errBase)) {
+            // It's the last model modem or a RMA is already done
+            const typeMessage = error.data.message.substring(0, 6);
```

Comment:
Can a `includes` be used instead ? Taking the first 6 characters looks random. What if the spec change and error is now based on the 7 first characters, we will have to update all, instead of just the constant mapping

Reply:
No, it's not. I have the list of type error defined that I can receive from the API called.